numpy/ctypeslib.py

```diff
@@ -97,10 +97,10 @@ def load_library(libname, loader_path):
 
         Parameters
         ----------
-        libname : str
+        libname : path-like
```

Comment:
Are paths with / actually allowed here? If not, this looks like it should remain a str

Reply:
It seems to be, yes.
What happens when defining `libpath` is, roughly speaking, calling  `os.path.join` on `loader_path` and `libname` the input parameters.

https://github.com/numpy/numpy/blob/4afd82d8dd2c1ef48c62bdf875a194e407f2d2d3/numpy/ctypeslib.py#L144-L148

Reply:
Hmm, to me that implies the path component should always be in `loader_path`, and `libname` should always be a string.

Reply:
Perhaps, though `os.path.join` is not required to join a path and a file name; 
the joining of two paths is also perfectly acceptable.

``` python
In [1]: import os

In [2]: os.path.join('a/b/c/', 'd/e/f')
Out[2]: 'a/b/c/d/e/f'
```

Besides, why should it remain a string if a / is not allowed?
The likes of `pathlib.Path` are just generally useful objects, regardless of whether it is used to represent a path- or file-name.